purchase_currency/models/sale_order_line.py

```diff
@@ -0,0 +1,16 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _purchase_service_prepare_order_values(self, supplierinfo):
+        result = super()._purchase_service_prepare_order_values(supplierinfo)
+        supplier_currency = supplierinfo.partner_id.property_purchase_currency_id
```

Comment:
Should we explicit `with_company(self.company_id)` to be extra sure to get the currect currency? `supplierinfo` should already have the company context from previous `_purchase_service_create`, maybe we should doublecheck to make sure

Reply:
> `_purchase_service_create`

The current company is taken from the sale.order.line record. I think is correct situation.